wagtail/contrib/table_block/tests.py

```diff
@@ -193,3 +200,163 @@ def test_render_with_extra_context(self):
         })
         self.assertIn("Test 1", result)
         self.assertIn("<div>A fascinating table.</div>", result)
+
+
+class TestTableBlockForm(WagtailTestUtils, SimpleTestCase):
+
+    def setUp(self):
+        # test value for table data
+        self.value = {
+            'first_row_is_table_header': True,
+            'first_col_is_header': True,
+            'data': [
+                ['Ship', 'Type', 'Status'],
+                ['Galactica', 'Battlestar', 'Active'],
+                ['Valkyrie', 'Battlestar', 'Destroyed'],
+                ['Cylon Basestar', 'Basestar', 'Active'],
+                ['Brenik', 'Small Military Vessel', 'Destroyed'],
+            ]
+        }
+        # set language from testing envrionment
+        language = translation.get_language()
+        if language is not None and len(language) > 2:
+            language = language[:2]
+
+        self.default_table_options = DEFAULT_TABLE_OPTIONS.copy()
+        self.default_table_options['language'] = language
+
+    def test_default_table_options(self):
+        """
+        Test options without any custom table_options provided.
+        """
+        block = TableBlock()
+        # check that default_table_options created correctly
+        self.assertEqual(block.table_options, block.get_table_options())
+        # check that default_table_options used on self
+        self.assertEqual(self.default_table_options, block.table_options)
+        # check a few individual keys from DEFAULT_TABLE_OPTIONS
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['startRows'], block.get_table_options()['startRows'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['startRows'], block.table_options['startRows'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['colHeaders'], block.get_table_options()['colHeaders'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['colHeaders'], block.table_options['colHeaders'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['contextMenu'], block.get_table_options()['contextMenu'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['contextMenu'], block.table_options['contextMenu'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['colHeaders'], block.get_table_options()['colHeaders'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['colHeaders'], block.table_options['colHeaders'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['stretchH'], block.get_table_options()['stretchH'])
+        self.assertEqual(DEFAULT_TABLE_OPTIONS['stretchH'], block.table_options['stretchH'])
```

Comment:
Checking both `block.get_table_options()` and `block.table_options` is a bit heavy, probably unnecessary\* and probably error prone. As a matter of fact, you check `colHeaders` twice (lines 240-241 and 244-245). Should the second occurrences be `rowHeaders`.

\* There is:
- `self.assertEqual(block.table_options, block.get_table_options())` above so checking either one should be enough
- `self.assertEqual(self.default_table_options, block.table_options)` already check that the defaults are conserved.